src/track/timesignature.h

```diff
@@ -0,0 +1,54 @@
+// signature.h
+// Created on: 06/11/2019 by Javier Vilarroig
+
+#pragma once
+
+namespace mixxx {
+
+/// Musical Time signature
+/// Right now is to be used for bar detection only, so only the beats per bar is useful
+class TimeSignature final {
+  private:
+    int m_beatsPerBar;
+    int m_noteValue;
+
+  public:
+    inline TimeSignature(int beatsPerBar, int noteValue)
```

Comment:
The implicit default constructor will leave the members uninitialized. Declare one (constexpr) that initializes them to 4/4.

Reply:
Should I use mixxx::track::io::TimeSignature as the underlying object for the TimeSignature as I have done with Beat class in #2844 ?
This would mean a protobuf generated object is the member variable instead of two integers.
The default (4/4) is already defined in `beats.proto`.

Reply:
Or delete the default constructor? If calling code should use a default value of 4/4, that should be made explicit in the calling code, not hidden in the implementation of the default constructor of this class.

Reply:
Or use the null 0/0 value for the default constuctor

Reply:
> Or use the null 0/0 value for the default constuctor

This would so be consistent with Qt classes where the default constructor creates an invalid value.